openquake/hazardlib/gsim/campbell_1997.py

```diff
@@ -0,0 +1,145 @@
+    REQUIRES_SITES_PARAMETERS = set(('vs30',))
+
+    #: Required rupture parameters are magnitude and top of rupture depth
+    REQUIRES_RUPTURE_PARAMETERS = set(('mag', 'rake'))
+
+    #: Required distance measure is closest distance to rupture
+    REQUIRES_DISTANCES = set(('rrup', ))
+
+    #: Verification of the mean value was done by digitizing Figs. 9 and 10
+    #: using Engauge Digitizer. The tests check varied magnitude, distance,
+    #: vs30, and faulting type. Maximum error was ~1.3%. OpenQuake trellis
+    #: plots match these figures. Also tested against a matlab implementation
+    #: (web.stanford.edu/~bakerjw/GMPEs/C_1997_horiz.m), which also has no
+    #: verification tables.
+
+    non_verified = True
+
+    def get_mean_and_stddevs(self, sites, rup, dists, imt, stddev_types):
+        """
+        See :meth:`superclass method
+        <.base.GroundShakingIntensityModel.get_mean_and_stddevs>`
+        for spec of input and result values.
+        """
+        R = (dists.rrup)
+        M = rup.mag
+        # get constants
+        Ssr = self.get_Ssr_term(sites.vs30)
+        Shr = self.get_Shr_term(sites.vs30)
+        rake = rup.rake
+        F = self.get_fault_term(rake)
+
+        # compute mean
+        mean = -3.512 + (0.904 * M) - (1.328 * np.log(np.sqrt(R**2
+               + (0.149 * np.exp(0.647 * M))**2))) \
+               + (1.125 - 0.112 * np.log(R) - 0.0957 * M) * F \
+               + (0.440 - 0.171 * np.log(R)) * Ssr \
+               + (0.405 - 0.222 * np.log(R)) * Shr
+        stddevs = self.get_stddevs(mean)
+        return mean, stddevs
+
+    def get_fault_term(self, rake):
+        """
+        Returns coefficient for faulting style (pg 156)
+        """
+        rake = rake + 360 if rake < 0 else rake
+
+        if (rake >= 45) & (rake <= 135):
+            f = 1.
+        elif (rake >= 225) & (rake <= 315):
+            f = 0.5
+        else:
+            f = 0.
+        return f
+
+    def get_Ssr_term(self, vs30):
+        """
+        Returns site term for soft rock (pg 157)
+        """
+        ssr = []
+        for v in vs30:
+            if (v >= 760.) & (v < 1500.):
+                ss = 1.
+            else:
+                ss = 0.
+            ssr.append(ss)
+        return ssr
+
+    def get_Shr_term(self, vs30):
+        """
+        Returns site term for hard rock (pg 157)
+        """
+        return vs30 >= 1500
+
+    def get_stddevs(self, mean):
+        """
+        Returns the standard deviations from mean (pg 164; more robust than
+        estimate using magnitude)
+        """
+        sig_n = np.zeros(len(mean))
+        stddevs = []
+        i = 0
+        for m in mean:
+            if np.exp(m) < 0.068:
+                sig = 0.55
+            elif (np.exp(m) >= 0.068) & (np.exp(m) <= 0.21):
+                sig = 0.173 - 0.140 * np.log(np.exp(m))
+            else:
+                sig = 0.39
+            sig_n[i] = (sig)
+            i += 1
+        stddevs.append(sig_n)
```

Comment:
Try this:
```python
def get_stddevs(self, mean, stddev_types):
    """
    Returns the ...
    """
    mean = np.exp(mean)
    sigma = 0.39 + np.zeros(mean.shape)
    sigma[mean < 0.068] = 0.55
    idx = np.logical_and(mean >= 0.068, mean <= 0.21)
    sigma[idx] = 0.173- 0.140 * np.log(mean[idx])
    stddevs = []
    for stddev in stddev_types:
        if stddev == const.StdDev.Total:
            stddevs.append(sigma)
    return stddevs
```

Then change invoking call accordinly.

Reply:
Okay, I changed to this